Assets/Plugins/Source/Core/Config.cs

```diff
@@ -83,26 +89,76 @@ protected Config(string filename, string directory)
+        protected static void RegisterFactor<T>(Func<Config> factory) where T : Config
+        {
+            s_factories[typeof(T)] = factory;
+        }
+
+        /// <summary>
+        /// Try to retrieve the factory method for the indicated type that can be used to
+        /// create a new instance of the given config type.
+        /// </summary>
+        /// <typeparam name="T">The type of config to get the factory method for.</typeparam>
+        /// <param name="factory">The factory method that instantiates the config indicated.</param>
+        /// <returns>True if the factory is registered, false otherwise.</returns>
+        /// <exception cref="InvalidOperationException">
+        /// If the indicated type does not have a corresponding constructor function
+        /// registered, a verbose exception will be thrown indicating how to properly
+        /// implement the Config implementing class such that it's constructor is
+        /// properly registered.
+        /// </exception>
+        private static bool TryGetFactory<T>(out Func<Config> factory) where T : Config
+        {
+            if(!s_factories.TryGetValue(typeof(T), out factory))
+            {
+                throw new InvalidOperationException(
+                    $"No factory method has been registered for type \"{typeof(T).FullName}\". " +
+                    $"Please make sure that \"{typeof(T).FullName}\" registers it's constructor with the base Config class via a static constructor.");
+            }
+
+            return true;
+        }
+
         /// <summary>
         /// Retrieves the indicated Config object, reading its values into memory.
         /// </summary>
         /// <typeparam name="T">The Config to retrieve.</typeparam>
         /// <returns>Task<typeparam name="T">Config type.</typeparam></returns>
-        public static async Task<T> GetAsync<T>() where T : Config, new()
+        public static async Task<T> GetAsync<T>() where T : Config
         {
+            // NOTE: This block (and the corresponding one below) exists so that
+            //       the config values are only cached when not in the editor.
+            //       In the editor, config files can be changed, so they should
+            //       not be cached.
 #if !UNITY_EDITOR
+            // Determine if there is a cached copy of the config, return that if it exists.
             if (s_cachedConfigs.TryGetValue(typeof(T), out Config config))
             {
                 return (T)config;
             }
 #endif
+            // Try to get the factory method used to instantiate the config.
+            TryGetFactory<T>(out Func<Config> factory);
+
+            // Use the factory method to create the config.
+            T instance = (T)factory();
 
-            T instance = new();
+            // Asynchronously read the config values from the corresponding file.
             await instance.ReadAsync();
```

Comment:
I can see how that could be misleading. The `Read` and `ReadAsync` functions do not actually return anything, they perhaps more appropriately could be named `ReadValuesFromFile` or something like that - because as the function comments explain:
```c#
/// <summary>
/// Synchronously reads the contents of a Config from the json file that defines it.
/// </summary>
protected virtual void Read() { /*...*/ }
```

The reason that the `ReadAsync` function returns a `Task` is that it is required to do so in-order to fulfill the `async` modifier - which makes the method `awaitable`. 

```c#
/// <summary>
/// Asynchronously read the values from the JSON file associated with this Config
/// </summary>
/// <returns>Task.</returns>
protected virtual async Task ReadAsync() { /*...*/ }
```

That said, within the context of the code the function names `ReadAsync` and `Read` seemed sufficiently descriptive and certainly more succinct than `ReadValuesFromFileIntoMemory` or something like that.